guides/messaging-high-availability.adoc

```diff
@@ -0,0 +1,227 @@
+= Configure WildFly With Messaging (ActiveMQ Artemis) Cluster And High Availability
+:summary: Learn how to configure WildFly with messaging (ActiveMQ Artemis) cluster and high availability
+:includedir: _includes
+include::{includedir}/_attributes.adoc[]
+// you can override any attributes eg to lengthen the
+// time to complete the guide
+:prerequisites-time: 20
+
+In this guide you will learn how to configure two WildFly servers with messaging (integrated ActiveMQ Artemis broker)
+in high availability topology with https://activemq.apache.org/components/artemis/documentation/latest/ha.html#shared-store[shared journal].
+
+The Apache ActiveMQ Artemis broker, embedded within WildFly, supports linking brokers together in primary-secondary pairs.
+This creates an active-passive HA design: under normal conditions, the primary broker handles all the workload, while the secondary node remains on standby,
+ready to take over if the primary broker fails. However, we will not start another WildFly server solely to support an inactive Artemis broker, instead we will configure an additional ActiveMQ Artemis broker
+on each WildFly instance to act as the secondary broker for the primary in the other WildFly. This setup results in two primary-secondary pairs, with each pair maintaining its own journal directory.
+
+image::artemis-collocated-topology.png[WildFly with ActiveMQ Artemis in collocated HA topology]
+
+If one WildFly server fails, the secondary ActiveMQ Artemis broker on the other WildFly server will activate and take over all duties.
+All HA enabled connections from the original primary broker will fail over to the secondary broker. The secondary broker will
+access all messages stored on the primary broker by loading them from the shared journal directory. Once the original WildFly server is restarted,
+the secondary broker will automatically switch back to standby mode, and all clients will fail back to the primary broker.
+
+[NOTE]
+--
+For simplicity, we will use just one machine to run WildFly instances and local directories for shared journals. In real world scenario those should be two separate machines with shared journal mount from shared file system (like NFS4).
```

Comment:
s/and local directories/and use local directories/g

The first 'use' verb is too far from the local directories part for it to be obvious that the local directories part is a 2nd object to the verb.

Reply:
s/In real/In a real/g
s/with shared/with a shared/g
s/mount from/mounted from a/g